rust/rope/src/compare.rs

```diff
@@ -405,9 +397,9 @@ mod tests {
         let two = "aaaa";
         let tre = "aaba";
         let fur = "";
-        assert!(ne_idx_sse42(one.as_bytes(), two.as_bytes()).is_none());
-        assert_eq!(ne_idx_sse42(one.as_bytes(), tre.as_bytes()), Some(2));
-        assert_eq!(ne_idx_sse42(one.as_bytes(), fur.as_bytes()), None);
+        assert!(ne_idx_sse(one.as_bytes(), two.as_bytes()).is_none());
```

Comment:
So many of these tests fail if you specify `RUSTFLAGS='-C target-cpu=native'` for example. I think what you want to do is this:

* Don't `#[cfg(target_feature)]`.
* Do `#[cfg(target_arch)]`.
* Do runtime `is_x86_feature_detected` check.
* Wrap call in `unsafe`.

Apply consistently to all such tests; I see some closer but none fully correct.

Would be good to also test on aarch64, but I don't have a system super handy. I could probably get rust compiler set up on my rpi3 though.